pandas/core/reshape/merge.py

```diff
@@ -1356,8 +1357,14 @@ def _get_merge_keys(self):
                 if self.tolerance < 0:
                     raise MergeError("tolerance must be positive")
 
+            elif is_float_dtype(lt):
```

Comment:
Can you just combine the integer and float dtype check? Looks like they have similar branches

Reply:
I see that they're similar but I think they're both needed so that we only allow integer tolerances for integer series and any numeric for float series. Unless you can think of a succinct way to combine

Reply:
A little bulky but:
```
elif is_numeric_dtype(lt):
    if (is_int64_dtype(lt) and not is_integer(self.tolerance)) or (is_float_dtype(lt) and not is_number(self.tolerance)):
        raise ...
    if self.tolerance < 0:
        raise ...
```

Reply:
I personally think that's less readable, I would rather keep it the way it is unless others also feel the same way